sql/_13_issues/_22_1h/cases/cbrd_24149.sql

```diff
@@ -0,0 +1,251 @@
+drop table if exists tbl1;
+
+create table tbl1 (var_col varchar(20), char_col char(20));
+create index var_idx on tbl1(var_col);
+create index char_idx on tbl1(char_col);
+create index midx_idx1 on tbl1(var_col,char_col);
+create index midx_idx2 on tbl1(char_col,var_col);
+create index var_idx_desc on tbl1(var_col desc);
+create index char_idx_desc on tbl1(char_col desc);
+create index midx_idx1_desc on tbl1(var_col desc,char_col desc);
+create index midx_idx2_desc on tbl1(char_col desc,var_col desc);
+
+insert into tbl1 values('a','a');
+insert into tbl1 values('a ','a ');
+insert into tbl1 values('a  ','a  ');
+insert into tbl1 values('a a','a a');
+insert into tbl1 values('ab','ab');
+insert into tbl1 values('ab ','ab ');
+insert into tbl1 values('a  b','a  b');
+insert into tbl1 values(NULL,NULL);
+
+
+-- char to char
+
+--heap scan
+select /*+ recompile */ char_col from tbl1 where char_col = 'a  ' using index none;
+
+--char index scan
+select /*+ recompile */ char_col from tbl1 where char_col = 'a  ' using index char_idx;
+select /*+ recompile */ char_col from tbl1 where char_col = 'a  ' using index char_idx_desc;
+
+--multi col index scan
+select /*+ recompile */ char_col from tbl1 where char_col = 'a  ' using index midx_idx2;
+select /*+ recompile */ char_col from tbl1 where char_col = 'a  ' using index midx_idx2_desc;
+select /*+ recompile index_ss */ char_col from tbl1 where char_col = 'a  ' using index midx_idx1;
+select /*+ recompile index_ss */ char_col from tbl1 where char_col = 'a  ' using index midx_idx1_desc;
+
+--join
+select /*+ recompile */ a.char_col from tbl1 a, tbl1 b where a.char_col = b.char_col;
+
+--table scan
+select /*+ recompile */ a.char_col from tbl1 a where a.char_col = a.char_col;
+
+-- in operation
+select /*+ recompile */ char_col from tbl1 where char_col in ('a  ','ab     ') using index none;
+select /*+ recompile */ char_col from tbl1 where char_col in ('a  ','ab     ') using index char_idx;
+select /*+ recompile */ char_col from tbl1 where char_col in ('a  ','ab     ')  using index midx_idx2;
+select /*+ recompile index_ss */ char_col from tbl1 where char_col in ('a  ','ab     ') using index midx_idx1;
+
+select /*+ recompile */ char_col from tbl1 where char_col in (char_col,char_col);
+select /*+ recompile */ char_col from tbl1 where (char_col,char_col) in ((char_col,char_col),(char_col,char_col));
+
+-- string function (decode)
+select /*+ recompile */ char_col, decode(char_col,'a',1,'a  b',2,'ab',3,4) from tbl1;
+
+-- prepare statement
+prepare c_stmt from 'select char_col as "prepare_c" from tbl1 where char_col = ?';
+execute c_stmt using 'a';
+execute c_stmt using 'a     ';
+drop prepare char_stmt;
```

Comment:
it should be changed to "drop prepare c_stmt" not to handle an error and its answer should also modified.

Reply:
fix it